applications/GeoMechanicsApplication/custom_elements/U_Pw_updated_lagrangian_FIC_element.cpp

```diff
@@ -162,40 +155,16 @@ template <unsigned int TDim, unsigned int TNumNodes>
 void UPwUpdatedLagrangianFICElement<TDim, TNumNodes>::CalculateOnIntegrationPoints(
     const Variable<Matrix>& rVariable, std::vector<Matrix>& rOutput, const ProcessInfo& rCurrentProcessInfo)
 {
-    // Defining necessary variables
-    const GeometryType& rGeom      = this->GetGeometry();
-    const unsigned int  NumGPoints = rGeom.IntegrationPointsNumber(mThisIntegrationMethod);
-
-    if (rOutput.size() != NumGPoints) rOutput.resize(NumGPoints);
+    rOutput.resize(this->GetGeometry().IntegrationPointsNumber(mThisIntegrationMethod));
```

Comment:
Anne, for my knowledge what is preferable to use clear() or resize()? Above you used clear() for doubles and here resize for Matrix.

Reply:
Good observation. The short answer is: that depends. For the Ublas containers (e.g. `Vector` or `Matrix`), we have little choice, since they don't support `push_back` or `emplace_back`. In those cases, we need to ensure that the desired size(s) is/are set upfront. For STL containers (e.g. `std::vector`), we can actually use operations to append new data. In general, I prefer that, although I don't have a solid motivation for it. I may have seen this in one of the books by Stroustrup, but I'm not sure.

The `clear` that you refer to was applied to a `std::vector` object, where the effect of the previous `resize` operation was limited in scope. In other cases, the `resize` is done at a higher level. As such, removing it there may have an effect on other parts of the code that I don't want to touch right now. Therefore, I have kept the `resize` operation in place, but I did remove the size check (which is already done by the `resize` operation).